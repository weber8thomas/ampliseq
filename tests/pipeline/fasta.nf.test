nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_fasta"
    tag "pipeline"

    test("Fasta Input") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/software_versions.yml").exists() },
                { assert snapshot(path("$outputDir/asv_length_filter/ASV_len_filt.tsv"),
                                path("$outputDir/asv_length_filter/ASV_len_orig.tsv"),
                                path("$outputDir/asv_length_filter/ASV_seqs.len.fasta"),
                                path("$outputDir/asv_length_filter/stats.len.tsv")).match("asv_length_filter") },
                { assert snapshot(path("$outputDir/barrnap/rrna.arc.gff"),
                                path("$outputDir/barrnap/rrna.bac.gff"),
                                path("$outputDir/barrnap/rrna.euk.gff"),
                                path("$outputDir/barrnap/rrna.mito.gff")).match("barrnap") },
                { assert snapshot(path("$outputDir/codon_filter/ASV_codon_filtered.fna"),
                                path("$outputDir/codon_filter/ASV_codon_filtered.list")).match("codon_filter") },
                { assert new File("$outputDir/barrnap/summary.tsv").exists() },
                { assert snapshot(path("$outputDir/dada2/ref_taxonomy.rdp_18.txt")).match("dada2") },
                { assert new File("$outputDir/dada2/ASV_tax_species.rdp_18.tsv").exists() },
                { assert new File("$outputDir/dada2/ASV_tax.rdp_18.tsv").exists() },
                { assert snapshot(path("$outputDir/input/ASV_seqs.fasta")).match("input") },
                { assert new File("$outputDir/summary_report/summary_report.html").exists() }
            )
        }
    }
}
