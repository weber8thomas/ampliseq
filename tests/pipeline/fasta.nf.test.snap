{
    "input": {
        "content": [
            "ASV_seqs.fasta:md5,39b0d4ad96b37fa68d4ee5c21b667fed"
        ],
        "timestamp": "2023-05-28T21:06:17+0000"
    },
    "dada2": {
        "content": [
            "ref_taxonomy.rdp_18.txt:md5,815c4fce9f3d1de019fb995a43fb66ed"
        ],
        "timestamp": "2023-05-28T21:06:17+0000"
    },
    "asv_length_filter": {
        "content": [
            "ASV_len_filt.tsv:md5,0e5bd974219951f56db71843053ac796",
            "ASV_len_orig.tsv:md5,09a300b31acf0eee817367dad5083f48",
            "ASV_seqs.len.fasta:md5,9479b62fcff165963b33821e03fe7d7e",
            "stats.len.tsv:md5,6cc39bcbd889634364499f3a5a82af35"
        ],
        "timestamp": "2023-08-31T13:06:17+0000"
    },
    "barrnap": {
        "content": [
            "rrna.arc.gff:md5,6dae470aace9293d5eb8c318584852dd",
            "rrna.bac.gff:md5,439a9084f089120f700f938dfb58fa41",
            "rrna.euk.gff:md5,c9bc1d9d8fb77dc19c95dee2d53840eb",
            "rrna.mito.gff:md5,df19e1b84ba6f691d20c72b397c88abf"
        ],
        "timestamp": "2023-05-28T21:06:17+0000"
    },
    "codon_filter": {
        "content": [
            "ASV_codon_filtered.fna:md5,9eac2d17e026ac8b1cba876ff3cac619",
            "ASV_codon_filtered.list:md5,6cdde55829be1ebc5d73a17602e7c881"
        ],
        "timestamp": "2023-08-31T13:06:17+0000"
    }
}
